msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,348 @@
+* Bots must respond to user commands within two seconds or display a typing indicator.
+* Messaging extensions must respond to user commands within five seconds.
+* Notification must load within five seconds of the user action.
 
-* [Authentication documentation](../../../authentication/authentication.md)
-* [Bot authentication sample in Node](https://github.com/OfficeDev/microsoft-teams-sample-auth-node)
-* [Tab authentication sample in Node](https://github.com/OfficeDev/microsoft-teams-sample-complete-node)
-* [Tab/bot authentication in C#/.NET](https://github.com/OfficeDev/microsoft-teams-sample-complete-csharp)
+## 4.0 App package and store listing
 
-### &#9989; Response times must be reasonable
+App packages must be correctly formatted and include all required information and components.
 
-* **Tabs**. If a response to an action takes more than three seconds, you must provide a loading message or warning.
-* **Bots**. A response to a user command must occur within two seconds. If longer processing is required, your app must display a typing indicator.
-* **Compose extensions**. A response to a user command must occur within five seconds.
+### 4.1 App manifest
+
+* The app manifest must conform to the [latest manifest schema](~/resources/schema/manifest-schema.md).
+* If your app includes a bot or messaging extension, the manifest must be consistent with Bot Framework metadata, including bot name, logo, privacy policy link, and terms of service link.
+* If your app uses Azure Active Directory (Azure AD) for authentication, include the Azure AD ID in the manifest. For more information, see [manifest schema](~/resources/schema/manifest-schema.md#webapplicationinfo).
+
+### 4.2 App icons
+
+* Your app package must include two PNG versions of your app icon: a color icon and an outline icon.
+* Correctly sized and formatted icons must be specified in your app package. The icons must match what's submitted with the store listing metadata. For more information, see the Teams app [icon guidelines](~/concepts/build-and-test/apps-package.md#app-icons).
+
+### 4.3 App description
+
+The app descriptions in your manifest and store must be the same and follow these guidelines:
+
+* Use "Microsoft Teams" on first reference. Subsequent references can be shortened to ”Teams”.
+* Do not abbreviate "Microsoft" (such as "MS" or "MSFT").
+* Avoid unnecessary capitalization and language errors.
+* Follow the [short description guidelines](~/concepts/deploy-and-publish/appsource/prepare/detail-page-checklist.md#short-description).
+* Follow the [long description guidelines](~/concepts/deploy-and-publish/appsource/prepare/detail-page-checklist.md#long-description).
+* Clearly describe any limitations, conditions, or exceptions to the functionality, features, and deliverables described in the listing and related materials before the customer acquires your offer. The capabilities you declare must relate to the core functions and description of your offer.
+
```

Comment:
The app descriptions in your manifest and Partner center submission must be the same. You need a short and long description of your app. 
App descriptions should not include references that directly or through insinuation disparage another brand (whether Microsoft owned or otherwise). Ensure your description does not include claims that cannot be substantiated (for example, "guaranteed 200% increase in efficiency").
Short description
•	A concise summary of your app that highlights value proposition and is directed at your target audience. Keep the short description to a single sentence. 
•	Put the most important information first.
•	Include keywords that customers are likely to search for.
•	Don’t repeat your application name 
•	Should not use “app” in the short description
Long description
The long description can provide an engaging narrative that highlights the app value proposition, business audience, and target industry. 
While this description can be as long as 4000 characters, most users will only read between 300-500 words.


Do’s | Don’t
-- | --
Clearly describe any        limitations, conditions, or exceptions to the functionality, features,        and deliverables described in the listing and related materials before        the customer acquires your offer. The capabilities you declare must        relate to the core functions and description of your offer.    Use Markdown to        format your description.    List        features with bullet points so it's easier to scan the description.    Use        active voice and speak to users directly (for example, You can ...).    Include        a help or support link.    Reference Microsoft product        names only when necessary. For more information on the guidelines,        see Microsoft        Trademark and Brand Guidelines.    Refer        to Microsoft        365 instead of Office 365.      Use   the following language to describe how the app works with Microsoft products:       If you need to reference Teams, write        the first reference as Microsoft        Teams. Subsequent references can be shortened to Teams.    …works with Microsoft Teams    …working with Microsoft Teams    …within Microsoft Teams    …for Microsoft Teams    The same terms above can be used        interchangeably with Microsoft 365. | Exceed 500 words.    Abbreviate        Microsoft as MS or MSFT.    Indicate        the app is an offering from Microsoft, including using Microsoft slogans        or taglines.    Use        copyrighted brand names you don't own.    Include typos, grammatical        errors, unnecessary capitalizations (e.g. Users instead        of users),        acronyms, jargons.    Must not contain        any links to AppSource within the description.          Don’t use the following   language, unless certified as our official partner -        ……integrated with Microsoft        Teams    …integrates with…    …built for…    ….built on…    ….runs on…    …enabled by…    …certified for…    …developed for…    …designed for…    Slogans and taglines Partners        may not use or imitate any Microsoft tagline, or use the name of any        Microsoft product or service in their slogan or tagline